src/Provider/AbstractProvider.php

```diff
@@ -307,16 +305,13 @@ public function userScreenName($response, AccessToken $token)
     }
 
     /**
-     * Throws an IDPException when an error response is received
+     * Throws an IdentityProviderException when an error response is received
      *
      * @param array $result
      *
-     * @throws IDPException
+     * @throws IdentityProviderException
      */
-    public function throwIDPException(array $result)
-    {
-        throw new IDPException($result);
-    }
+    abstract public function errorCheck(array $result);
```

Comment:
I have used two oauth2 service and they all have their custom error response. So maybe it is hard to provider a default errorCheck suitable for most cases. And more important, if we provider a default behavior, third party provider authors will tend to be less concerned about error check.  Once the default behavior isn't suitable and the author overlooks it, the end user will be harmed. So in my opinion, being abstract is better. It enforces paying attention to error check.